pages/reviewer/ReviewerHandler.php

```diff
@@ -88,9 +90,14 @@ protected function _validateAccessKey($request)
             return;
         }
 
-        $reviewerSubmissionDao = DAORegistry::getDAO('ReviewerSubmissionDAO'); /** @var ReviewerSubmissionDAO $reviewerSubmissionDao */
-        $reviewerSubmission = $reviewerSubmissionDao->getReviewerSubmission($reviewId);
-        if (!$reviewerSubmission) {
+        $reviewAssignmentDao = DAORegistry::getDAO('ReviewAssignmentDAO'); /** @var ReviewAssignmentDAO $reviewAssignmentDao */
+        $reviewAssignment = $reviewAssignmentDao->getById($reviewId);
+        if (!$reviewAssignment) {
+            return;
+        } // e.g. deleted review assignment
+
+        $reviewSubmission = Repo::submission()->get($reviewAssignment->getSubmissionId());
```

Comment:
Get the submission from the authorized context objects to make sure the user is authorized to access it:

```php
$reviewSubmission = $this->getAuthorizedContextObject(ASSOC_TYPE_SUBMISSION);
```

For extra protection, you can check that the submission ID matches the submission id of the review assignment.

Reply:
Changed it. I added another check like so 

```
$reviewSubmission = $this->getAuthorizedContextObject(Application::ASSOC_TYPE_SUBMISSION);
$reviewAssignmentSubmission = Repo::submission()->get($reviewAssignment->getSubmissionId());
if (!$reviewSubmission || ($reviewSubmission->getId() != $reviewAssignmentSubmission->getId())) {
   return;
} // e.g. deleted review assignment
```

Reply:
You don't need the second line. You should be able to do just:

```php
$reviewSubmission = $this->getAuthorizedContextObject(Application::ASSOC_TYPE_SUBMISSION);
if (!$reviewSubmission || $reviewSubmission->getId() != $reviewAssignment->getSubmissionId()) {
   return;
}
```

Reply:
shouldn't we make sure that `$reviewAssignment` submission exists?

Reply:
If the ID is identical then `$this->getAuthorizedContextObject(Application::ASSOC_TYPE_SUBMISSION);` already checks that it exists.

Reply:
ok fixed